trino/trino_test.go

```diff
@@ -765,6 +765,106 @@ func TestQueryColumns(t *testing.T) {
 	assert.Equal(t, actualTypes, expectedTypes)
 }
 
+func TestParametricDateTime(t *testing.T) {
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	rows, err := db.Query(`SELECT
+  cast(current_time as time(6)) AS timep,
+  cast(current_time as time(6) with time zone) AS timeptz,
+  cast(current_time as timestamp(6)) AS tsp,
+  cast(current_time as timestamp(6) with time zone) AS tsptz`,
+		sql.Named("X-Trino-Client-Capabilities", "PARAMETRIC_DATETIME"))
+	require.NoError(t, err, "Failed executing query")
+	assert.NotNil(t, rows)
+
+	columns, err := rows.Columns()
+	require.NoError(t, err, "Failed reading result columns")
+
+	assert.Equal(t, 4, len(columns), "Expected 4 result column")
```

Comment:
Is this necessary? Wouldn't `assert.Equal` called on a slice verify this?

Reply:
I just copied the test code from another test :-)

Reply:
We build another slice later on using that length, so I guess it's safer to verify it...